From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Mon, 22 Dec 2025 15:08:38 +0100
Subject: [PATCH] agent helper: Send standard error to journal

Otherwise we will most likely get the default of
`StandardError=inherit`, causing any log messages to be sent to the
peer, which gets confused and fails the authentication session.

See: https://github.com/polkit-org/polkit/issues/622
See: https://gitlab.archlinux.org/archlinux/packaging/packages/polkit/-/issues/4
---
 data/polkit-agent-helper@.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/polkit-agent-helper@.service.in b/data/polkit-agent-helper@.service.in
index e5ce65d9262d..0e1e5a4b012a 100644
--- a/data/polkit-agent-helper@.service.in
+++ b/data/polkit-agent-helper@.service.in
@@ -10,6 +10,7 @@ ExecStart=@libprivdir@/polkit-agent-helper-1 --socket-activated
 SuccessExitStatus=2
 StandardInput=socket
 StandardOutput=socket
+StandardError=journal
 LimitMEMLOCK=0
 LockPersonality=yes
 MemoryDenyWriteExecute=yes
